File: iot/listener.py
import paho.mqtt.client as mqtt

BROKER = "localhost"
PORT = 1883


def on_connect(client, userdata, flags, rc):
    print("Listener connesso al broker MQTT")
    client.subscribe("bins/+/fill_level")


def on_message(client, userdata, msg):
    topic = msg.topic
    payload = msg.payload.decode()

    print(f"Ricevuto: {topic} -> {payload}%")


client = mqtt.Client()
client.on_connect = on_connect
client.on_message = on_message

client.connect(BROKER, PORT, 60)
client.loop_forever()

File: iot/listener.js
const mqtt = require('mqtt');

const client = mqtt.connect('mqtt://localhost:1883');

client.on('connect', () => {
  console.log('Listener connesso al broker MQTT');

  client.subscribe('bins/+/fill_level', (err) => {
    if (err) {
      console.error('Errore subscription:', err);
    } else {
      console.log('Sottoscritto a bins/+/fill_level');
    }
  });
});

client.on('message', (topic, message) => {
  const payload = message.toString();

  console.log(`Ricevuto: ${topic} -> ${payload}%`);
});
